pkg/controllers/v1alpha1/dataload/implement.go

```diff
@@ -344,11 +358,52 @@ func (r *DataLoadReconcilerImplement) reconcileFailedDataLoad(ctx reconcileReque
 		return utils.RequeueIfError(err)
 	}
 
-	// 2. record event and no requeue
+	// 2. record event
 	log.Info("DataLoad failed, won't requeue")
 	jobName := utils.GetDataLoadJobName(utils.GetDataLoadReleaseName(ctx.Name))
 	r.Recorder.Eventf(&ctx.DataLoad, v1.EventTypeNormal, common.DataLoadJobFailed, "DataLoad job %s failed", jobName)
-	return utils.NoRequeue()
+
+	// 3. update the phase of the dataload to Finished and requeue
+	dataloadToUpdata := ctx.DataLoad.DeepCopy()
+	dataloadToUpdata.Status.Phase = cdataload.DataLoadPhaseFailedAndFinished
+	if err := r.Status().Update(context.TODO(), dataloadToUpdata); err != nil {
+		log.Error(err, "failed to updata the dataload")
+		return utils.RequeueIfError(err)
+	}
+	log.V(1).Info("Updata phase of the dataload to Finished successfully")
+	return utils.RequeueImmediately()
+}
+
+// reconcileFinishedDataLoad reconciles DataLoads that are in `DataLoadPhaseFinished` phase
+// in order to delete DataLoad after TTL
+func (r *DataLoadReconcilerImplement) reconcileFinishedDataLoad(ctx reconcileRequestContext) (ctrl.Result, error) {
```

Comment:
Maybe we can wrap logic here to another func but not a `reconcileFinished` one, and reuse the func in both `reconcileLoaded` and `reconcileFailed`.

Reply:
There are some logic codes that should be executed exactly once after the job finished. In order to avoid executing these codes repeatedly during reconciling dataload deletion, I seperate the logic which concerns about dataload deletion and wrap it into the function "reconcileFinishedDataLoad".